### PR TITLE
[Fix]CallKit ringing edge cases for already-handled incoming calls

### PR DESCRIPTION
**Pull request by ipavlidakis**
### 🔗 Issue Links

Resolves https://linear.app/stream/issue/IOS-1628/bughandle-edge-cases-for-callkit-ringing

### 🎯 Goal

Ensure CallKit stops ringing cleanly when an incoming call was already
handled elsewhere, and preserve the explicit backend leave reason for those
cases.

### 📝 Summary

- detect already-handled incoming calls when the current user already
  accepted, rejected, or missed the call
- detect calls that already ended elsewhere or were rejected by the other
  participants before the local device answers
- propagate explicit leave reasons through the CallKit end-call flow so the
  backend receives the correct rejection reason

### 🛠 Implementation

`CallKitService.checkIfCallWasHandled` now returns an optional leave reason
instead of a boolean, so the incoming-call flow can distinguish why the call
should stop ringing.

When `reportIncomingCall` fetches a call state that has already been handled,
it immediately ends the CallKit call and forwards that explicit leave reason.
`CallKitService.callEnded` now stores the leave reason on the pending
`CallEntry`, and `provider(_:perform:)` for `CXEndCallAction` prefers that
stored reason before falling back to `rejectionReasonProvider`.

### 🧪 Manual Testing Notes

1. Have another user call your device.
2. Immediately after the outgoing ringing starts, have the caller hang up.
3. Verify the CallKit notification appears on your device and disappears immediately.
4. Verify the call does not remain stuck in ringing state in the app.

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [x] This change should receive manual QA
- [x] Changelog is updated with client-facing changes
- [x] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Affected documentation updated (tutorial, CMS)

## Summary by CodeRabbit

* **Bug Fixes**
  * Resolved CallKit ringing edge cases: incoming calls already handled elsewhere are now ended reliably and rejection reasons are preserved and forwarded so call outcomes (answered, rejected, missed, ended, auto-leave) are reported consistently.
* **Tests**
  * Expanded and improved test coverage for incoming-call handling, leave/reject reasons, missed/rejected scenarios, and auto-leave behavior.

### CODE REVIEW

File: Sources/StreamVideo/CallKit/CallKitService.swift [hunk 1]
```diff
@@ -639,32 +658,110 @@ open class CallKitService: NSObject, CXProviderDelegate, @unchecked Sendable {
...
+    ///    another device.
+    /// 3. The creator rejected it.
+    /// 4. Every remaining participant except the current user and creator
+    ///    rejected it.
+    ///
+    /// - Parameter callState: The latest call state fetched from the backend
+    ///   before starting the local ringing timer.
+    /// - Returns: A backend leave reason when the call should stop ringing
+    ///   immediately, or `nil` when the local device should keep ringing.
+    ///   The current mapping is:
```
Comment (martinmitrevski) [inline Sources/StreamVideo/CallKit/CallKitService.swift]: is this mapping aligned with other SDKs?

Comment (ipavlidakis) [reply Sources/StreamVideo/CallKit/CallKitService.swift]: Now they are :) Thanks

**Review by martinmitrevski — APPROVED** [review]
